File: TPII-2025-2/PadraoGoF/Estrutural/Adapter/Adapter01.py
# 1 - Interface do Cliente
class AplicacaoCliente:
    def request(self):
        pass
class Cliente:
    def __init__(self, target):
        self.target = target

    def criaRequest(self):
        print("Cliente - Fazendo uma requisição")
        self.target.request()

# 2 - Serviço Existente
class Adaptee:
    def especificaRequest(self):
        print("Requisição especifica do Adaptee")

# 3 - Adapter 
class Adapter:
    def __init__(self, adaptee):
        self.adaptee = adaptee

    def request(self):
        self.adaptee.especificaRequest()

# 4 - Utilizando o Adaptar
adaptee = Adaptee()
adapter = Adapter(adaptee)
cliente = Cliente(adapter)

# Fazendo uma requisição - Requisição específica do Adaptee
cliente.criaRequest()

File: TPII-2025-2/PadraoGoF/Estrutural/Adapter/Adapter01.js
// 1 - Interface do cliente
class AplicacaoCliente{
    request() {}
}

class Cliente{
    constructor(target){
        this.target = target;
    }

    criaRequest(){
        console.log("Cliente - Fazendo uma requisição");
        this.target = this.target;
    }
}

// 2 - Serviço existente
class Adaptee{
    especificaRequest(){
        console.log("Requisição especifica do Adaptee")
    }
}

// 3 - Adapter
class Adapter{
    constructor(adaptee){
        this.adaptee = adaptee;
    }

    request(){
        this.adaptee.especificaRequest();
    }
}

// 4 - Utilizando o adapter
const adaptee = new Adaptee();
const adapter = new Adaptee(adaptee);
const cliente = new Cliente(adapter);

 //Fazendo uma requisição - Requisição especifica do Adaptee:
 cliente.criaRequest();